exonum/src/runtime/rust/service.rs

```diff
@@ -92,7 +92,20 @@ pub trait Service: ServiceDispatcher + Debug + 'static {
     /// act similarly on different nodes. In other words, the service should only use data available
     /// in the provided `CallContext`.
     ///
+    /// Note that if service was added within genesis block, it will be activated **immediately** and
+    /// thus `after_transactions` will be invoked for such a service after the genesis block creation.
+    /// If you aren't interested in the invocation for the genesis block, consider using
+    /// [`skip_for_genesis`] macro.
+    ///
+    /// Also note that invocation of [`blockchain::Schema::height`] will **panic** if invoked within
+    /// `after_transactions` of the genesis block. If you aren't going to skip the invocation for genesis
+    /// block and need to know current height, use [`blockchain::Schema::next_height`] instead.
```

Comment:
Nit: Awkward wording; `next_height` does not return the *current* height. I'd replace with

> If you aren't going to skip the handler for the genesis block, use `blockchain::Schema::next_height` to infer the current blockchain height.